rfc/0012-invalid-values/README.md

```diff
@@ -0,0 +1,107 @@
+```
+block MyInvalidRowFilter oftype InvalidTableRowFilter {}
+```
+
+#### Block-Type `InvalidTableCellReplacer`
+This block-type allows replacing all invalid values of a column with a valid value.
+```
+block MyInvalidTemperatureReplacer oftype InvalidTableCellReplacer {
+  column: "Temperature";  // refers to table column
+  value: 21;              // expression producing valid value
+}
+```
+
+### Sink Behavior
+In the sinks, we have to reflect the presence of the `INVALID` value type as well. Each sink should bring a default behavior in this regard and allow further configuration modes. The following is an example on how the SQLiteLoader might work.
+
+```
+block CarsLoader oftype SQLiteLoader {
+  table: "Cars";
+  file: "./cars.sqlite";
+  invalidValueStrategy: "meta-column-invalid"
+}
+```
+
+The newly introduced field `invalidValueStrategy` can have the values `"meta-table-mirrored"`, `"meta-table-linked"`, or `"ignore-row"`.
+
+
+#### Meta Table Mirrored
+An additional table is created with the naming convention `<value table name>_META`. In this table, there are columns that mirror the structure of the original table. For each column in the value table, we introduce three meta columns:
+- `<value column name>_isValid`, flag whether the value in the value column is valid or not, 
+- `<value column name>_details`, text for describing how the invalid value emerged,
+- `<value column name>_lastValidValue`, the last valid value before turning invalid. 
+
+Invalid values in the value table are assigned to `null`. As an implication, all value columns must be `nullable`.
+
+In the future, we might add more meta columns in this manner to the meta table.
+
+#### Meta Table Linked
+An additional table is created with the naming convention `<value table name>_META`. In this table, there are columns that mirror the structure of the original table. For each column in the value table, we introduce three meta columns:
+- `<value column name>_isValid`, flag whether the value in the value column is valid or not, 
+- `<value column name>_details`, text for describing how the invalid value emerged,
+- `<value column name>_lastValidValue`, the last valid value before turning invalid. 
+
+Invalid values in the value table are assigned to `null`. As an implication, all value columns must be `nullable`.
+
+In the future, we might add more meta columns in this manner to the meta table.
+
+#### Ignore Row
+This configuration ignores rows with invalid values. They are not written into the database. Ignored values are logged on the info level (not warning since actively configured).
+
+This resembles the current configuration.
+
+
+## Drawbacks
+- The `"meta-column"` configuration of the `SQLiteLoaderSQLiteLoader` is not really extensible for further meta data.
+
+## Alternatives
+- Don't introduce a concept for invalid values at all.
+- Move handling of invalid values to value type concept instead of using blocks.
```

Comment:
Here, I could think of a "invalid value safety" concept in our type system, similar to to "null safety" in modern programming languages like [Dart](https://dart.dev/null-safety) and [Kotlin](https://kotlinlang.org/docs/null-safety.html#nullable-types-and-non-null-types). E.g. having a `?` operator to turn a type into one that is able to hold `INVALID`, like `integer?`.

Reply:
Added it to Possible Future Enhancements.